### PR TITLE
feat: Merge if statements

### PR DESCRIPTION
**Pull request by bcollamore**

### CODE REVIEW

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/MergeIfStatementsAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+			}
+
+			// Parent has an else clause
+			if (parentIfSyntax.Else != null)
+			{
+				return;
+			}
+
+			// Has ||
+			if (ifStatementSyntax.Condition.DescendantTokens().Any((token) => { return token.Kind() == SyntaxKind.BarBarToken; }))
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/MergeIfStatementsAnalyzer.cs:91]: Consider making method bool IsBarBarToken(SyntaxToken token) to symplify this Linq syntax.

File: Philips.CodeAnalysis.Test/Maintainability/Maintainability/MergeIfStatementsAnalyzerTest.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		{
+			return new MergeIfStatementsAnalyzer();
+		}
+
+
+		[DataTestMethod]
+		[DataRow(@"if (1==1) { if(1==1) {} else {} }")]			// Has else clause
+		[DataRow(@"if (1==1) { int x; if (1==1) {} }")]         // Has multiple statements
+		[DataRow(@"if (1==1) { if (1==1) {} ; int x}")]         // Has multiple statements
+		[DataRow(@"{ if (1==1) {} }")]							// Parent is not if statement
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Test/Maintainability/Maintainability/MergeIfStatementsAnalyzerTest.cs:61]: Consider assigning the comment text to `DisplayName`

**Review by ynsehoornenborg — APPROVED** [review]

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/MergeIfStatementsAnalyzer.cs (hunk 1 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/MergeIfStatementsAnalyzer.cs]: done

File: Philips.CodeAnalysis.Test/Maintainability/Maintainability/MergeIfStatementsAnalyzerTest.cs (hunk 2 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.Test/Maintainability/Maintainability/MergeIfStatementsAnalyzerTest.cs]: done, and created Issue for new analyzer idea

**Review by ynsehoornenborg — APPROVED** [review]

**Pull request merged**